lib/heap_sort.rb

```diff
@@ -1,8 +1,25 @@
-
+require_relative 'min_heap'
 
 # This method uses a heap to sort an array.
-# Time Complexity:  ?
-# Space Complexity: ?
-def heap_sort(list)
-  raise NotImplementedError, "Method not implemented yet..."
-end
\ No newline at end of file
+# Time Complexity:  O(n); it touches each element in the list to make the heap and 
+# then again to remove the sorted elements and output them as a list O(2n), drop the constant.
+# Space Complexity: O(1); I think this is constant, because the heap is created as the array is
+# reduced so there is not a growth in memory space (except that a heap may take more space?).
+def heapsort(list)
```

Comment:
Good insight on space complexity, but the array doesn't actually shrink when you do `.pop` just the amount of spaces used inside it.  Otherwise this works.